src/targets/gpu/hip.cpp

```diff
@@ -183,8 +183,8 @@ argument register_on_gpu(const argument& arg)
 {
     auto arg_shared = arg.share();
     auto p          = register_on_gpu(arg_shared.data(), arg_shared.get_shape().bytes());
-    return {arg_shared.get_shape(),
-            [p, a = std::move(arg_shared)]() mutable { return get_device_ptr(p.get()); }};
+    auto s          = arg_shared.get_shape();
```

Comment:
Why is this assigned to another variable?

Reply:
Fixing issue caught by clang-tidy with possible ordering resolution of operands
due to std::move. Just captured the shape before arg_shared is used in the lambda

More details here.

https://releases.llvm.org/11.1.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-use-after-move.html